runtime/doc/lua-guide.txt

```diff
@@ -0,0 +1,311 @@
+	vim.opt.completeopt = {'menuone', 'noselect'}
+	print(vim.inspect(vim.opt.completeopt:get())) -- { "menuone", "noselect" }
+
+Wrappers for list-like, map-like and set-like options also come with methods
+and metamethods that work similarly to their |:set+=|, |:set^=| and |:set-=|
+counterparts in Vim script. >lua
+
+	vim.opt.shortmess:append({ I = true })
+	-- alternative form:
+	vim.opt.shortmess = vim.opt.shortmess + { I = true }
+
+	vim.opt.whichwrap:remove({ 'b', 's' })
+	-- alternative form:
+	vim.opt.whichwrap = vim.opt.whichwrap - { 'b', 's' }
+
+Be sure to look at |vim.opt| for more information.
+
+==============================================================================
+Defining mappings
+
+Neovim provides two functions to set and delete mappings, |vim.keymap.set()|
+and |vim.keymap.del()|.
+
+vim.keymap.set()
+-------------------------------------------------------------------------------
+
+|vim.keymap.set()| is a convenience function of API functions
+vim.api.|nvim_set_keymap()| and vim.api.|nvim_buf_set_keymap()|
+
+The first argument is a string or a table of strings containing the mode
+prefix for which the mapping will take effect: >lua
+
+	-- Normal mode
+	vim.keymap.set('n', '<Leader>ex1', '<Cmd>lua vim.notify("Example 1")<CR>')
+	-- Normal and Command-line mode
+	vim.keymap.set({'n', 'c'}, '<Leader>ex2', '<Cmd>lua vim.notify("Example 2")<CR>')
+
+The prefixes are the ones listed in |:map-arguments|, as well as "!" and and
+empty string "":
+
+| String value           | Help page     | Affected modes                           | Vim script equivalent |
+| ---------------------- | ------------- | ---------------------------------------- | --------------------- |
+| "" (empty string)      | |mapmode-nvo|   | Normal, Visual, Select, Operator-pending | |:map|                  |
+| `"n"`                    | |mapmode-n|     | Normal                                   | |:nmap|                 |
+| `"v"`                    | |mapmode-v|     | Visual and Select                        | |:vmap|                 |
+| `"s"`                    | |mapmode-s|     | Select                                   | |:smap|                 |
+| `"x"`                    | |mapmode-x|     | Visual                                   | |:xmap|                 |
+| `"o"`                    | |mapmode-o|     | Operator-pending                         | |:omap|                 |
+| `"!"`                    | |mapmode-ic|    | Insert and Command-line                  | |:map!|                 |
+| `"i"`                    | |mapmode-i|     | Insert                                   | |:imap|                 |
+| `"l"`                    | |mapmode-l|     | Insert, Command-line, Lang-Arg           | |:lmap|                 |
+| `"c"`                    | |mapmode-c|     | Command-line                             | |:cmap|                 |
+| `"t"`                    | |mapmode-t|     | Terminal                                 | |:tmap|                 |
```

Comment:
The non-empty strings have backticks but the empty string does not. I'd suggest removing these backticks.

Reply:
Do we want to duplicate Vim manual information? I'd say focus on the Lua specifics and link to the corresponding Vim help for basic information. (I don't think the goal is to give an introduction to Vim _and_ Neovim's Lua at the same time.)

Reply:
Yeah, was thinking this table is a bit much but decided last minute to not remove it. Will a simple

"The prefixes are the ones listed in |:map-modes|, or "!" for |:map!|, or empty fstring for |:map|."

suffice?